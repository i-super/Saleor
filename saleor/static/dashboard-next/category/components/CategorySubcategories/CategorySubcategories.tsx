import AddIcon from "@material-ui/icons/Add";
import Card from "material-ui/Card";
import IconButton from "material-ui/IconButton";
import * as React from "react";

import CategoryList from "../../../components/CategoryList";
import PageHeader from "../../../components/PageHeader";
import { CategoryPropertiesQuery } from "../../../gql-types";
import i18n from "../../../i18n";

interface CategorySubcategoriesProps {
  subcategories?: Array<{
    id: string;
    name: string;
  }>;
  onClickSubcategory?(id: string);
  onCreate?();
}

const CategorySubcategories: React.StatelessComponent<
  CategorySubcategoriesProps
> = ({ subcategories, onClickSubcategory, onCreate }) => (
  <Card>
    <PageHeader
      title={i18n.t("Subcategories", {
        context: "title"
      })}
    >
      <IconButton onClick={onCreate}>
        <AddIcon />
      </IconButton>
    </PageHeader>
    <CategoryList categories={subcategories} onClick={onClickSubcategory} />
  </Card>
);

export default CategorySubcategories;
