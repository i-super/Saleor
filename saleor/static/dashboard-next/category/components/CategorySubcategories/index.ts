export { default } from "./CategorySubcategories";
export * from "./CategorySubcategories";
