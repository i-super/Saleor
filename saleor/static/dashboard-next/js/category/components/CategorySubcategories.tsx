import AddIcon from "material-ui-icons/Add";
import Card from "material-ui/Card";
import IconButton from "material-ui/IconButton";
import * as React from "react";

import CategoryList from "../../components/CategoryList";
import PageHeader from "../../components/PageHeader";
import { CategoryPropertiesQuery } from "../../gql-types";
import i18n from "../../i18n";

interface CategorySubcategoriesProps {
  data: CategoryPropertiesQuery;
  loading?: boolean;
  onClickSubcategory?(id: string);
  onCreate?();
}

const CategorySubcategories: React.StatelessComponent<
  CategorySubcategoriesProps
> = ({ data, loading, onClickSubcategory, onCreate }) => (
  <Card>
    <PageHeader
      title={i18n.t("Subcategories", {
        context: "title"
      })}
    >
      <IconButton disabled={loading} onClick={onCreate}>
        <AddIcon />
      </IconButton>
    </PageHeader>
    <CategoryList
      categories={
        data.category && data.category.children && data.category.children.edges
      }
      onClick={onClickSubcategory}
    />
  </Card>
);

export default CategorySubcategories;
