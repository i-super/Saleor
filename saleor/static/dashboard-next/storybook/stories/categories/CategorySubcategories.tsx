import { storiesOf } from "@storybook/react";
import { DialogContentText } from "material-ui/Dialog";
import * as React from "react";

import CategorySubcategories from "../../../category/components/CategorySubcategories";

const categories = [
  {
    id: "123123",
    name: "Lorem ipsum dolor"
  },
  {
    id: "876752",
    name: "Mauris vehicula tortor vulputate"
  }
];

storiesOf("Categories / CategorySubcategories", module)
  .add("with initial data", () => (
    <CategorySubcategories
      subcategories={categories}
      onClickSubcategory={() => {}}
      onCreate={() => {}}
    />
  ))
  .add("without initial data", () => (
    <CategorySubcategories
      subcategories={[]}
      onClickSubcategory={() => {}}
      onCreate={() => {}}
    />
  ))
  .add("when loading data", () => (
    <CategorySubcategories onClickSubcategory={() => {}} onCreate={() => {}} />
  ));
