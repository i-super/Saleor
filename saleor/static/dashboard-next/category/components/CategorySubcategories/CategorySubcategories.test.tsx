import * as React from "react";
import { MemoryRouter } from "react-router-dom";
import * as renderer from "react-test-renderer";

import CategorySubcategories from "./CategorySubcategories";

describe("<CategorySubcategories />", () => {
  it("renders with initial data", () => {
    const categories = [
      {
        id: "123123",
        name: "Lorem ipsum dolor"
      },
      {
        id: "876752",
        name: "Mauris vehicula tortor vulputate"
      }
    ];
    const component = renderer.create(
      <CategorySubcategories
        subcategories={categories}
        onClickSubcategory={jest.fn()}
        onCreate={jest.fn()}
      />
    );
    expect(component).toMatchSnapshot();
  });
  it("renders without initial data", () => {
    const component = renderer.create(
      <CategorySubcategories
        subcategories={[]}
        onClickSubcategory={jest.fn()}
        onCreate={jest.fn()}
      />
    );
    expect(component).toMatchSnapshot();
  });
  it("renders when loading data", () => {
    const component = renderer.create(
      <CategorySubcategories
        onClickSubcategory={jest.fn()}
        onCreate={jest.fn()}
      />
    );
    expect(component).toMatchSnapshot();
  });
});
