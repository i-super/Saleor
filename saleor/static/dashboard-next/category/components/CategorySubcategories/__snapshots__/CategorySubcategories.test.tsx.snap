// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CategorySubcategories /> renders when loading data 1`] = `
<div
  className="MuiPaper-root-1 MuiPaper-elevation2-5 MuiPaper-rounded-2"
>
  <div
    className="Component-root-31"
  >
    <h2
      className="MuiTypography-root-34 MuiTypography-title-40 Component-title-33"
    >
      Subcategories
    </h2>
    <div
      className="Component-action-28"
    >
      <button
        className="MuiButtonBase-root-64 MuiIconButton-root-58"
        disabled={false}
        onBlur={[Function]}
        onClick={[MockFunction]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex="0"
        type="button"
      >
        <span
          className="MuiIconButton-label-63"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root-67"
            color={undefined}
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
            />
          </svg>
        </span>
        <span
          className="MuiTouchRipple-root-73"
        />
      </button>
    </div>
  </div>
  <ul
    className="MuiList-root-80 MuiList-padding-81"
  >
    <li
      className="MuiListItem-root-84 MuiListItem-default-87 MuiListItem-gutters-91"
      disabled={false}
    >
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root-67 MuiListItemIcon-root-94"
        color={undefined}
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M10 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V8c0-1.1-.9-2-2-2h-8l-2-2z"
        />
      </svg>
      <div
        className="MuiListItemText-root-95"
      >
        <h3
          className="MuiTypography-root-34 MuiTypography-subheading-41 MuiListItemText-primary-98"
        >
          <span
            className="Component-skeleton-106"
            style={undefined}
          >
            â€Œ
          </span>
        </h3>
      </div>
    </li>
  </ul>
</div>
`;

exports[`<CategorySubcategories /> renders with initial data 1`] = `
<div
  className="MuiPaper-root-1 MuiPaper-elevation2-5 MuiPaper-rounded-2"
>
  <div
    className="Component-root-31"
  >
    <h2
      className="MuiTypography-root-34 MuiTypography-title-40 Component-title-33"
    >
      Subcategories
    </h2>
    <div
      className="Component-action-28"
    >
      <button
        className="MuiButtonBase-root-64 MuiIconButton-root-58"
        disabled={false}
        onBlur={[Function]}
        onClick={[MockFunction]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex="0"
        type="button"
      >
        <span
          className="MuiIconButton-label-63"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root-67"
            color={undefined}
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
            />
          </svg>
        </span>
        <span
          className="MuiTouchRipple-root-73"
        />
      </button>
    </div>
  </div>
  <ul
    className="MuiList-root-80 MuiList-padding-81"
  >
    <div
      className="MuiButtonBase-root-64 MuiListItem-root-84 MuiListItem-default-87 MuiListItem-gutters-91 MuiListItem-button-92"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      tabIndex="0"
    >
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root-67 MuiListItemIcon-root-94"
        color={undefined}
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M10 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V8c0-1.1-.9-2-2-2h-8l-2-2z"
        />
      </svg>
      <div
        className="MuiListItemText-root-95"
      >
        <h3
          className="MuiTypography-root-34 MuiTypography-subheading-41 MuiListItemText-primary-98"
        >
          Lorem ipsum dolor
        </h3>
      </div>
      <span
        className="MuiTouchRipple-root-73"
      />
    </div>
    <div
      className="MuiButtonBase-root-64 MuiListItem-root-84 MuiListItem-default-87 MuiListItem-gutters-91 MuiListItem-button-92"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      tabIndex="0"
    >
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root-67 MuiListItemIcon-root-94"
        color={undefined}
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M10 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V8c0-1.1-.9-2-2-2h-8l-2-2z"
        />
      </svg>
      <div
        className="MuiListItemText-root-95"
      >
        <h3
          className="MuiTypography-root-34 MuiTypography-subheading-41 MuiListItemText-primary-98"
        >
          Mauris vehicula tortor vulputate
        </h3>
      </div>
      <span
        className="MuiTouchRipple-root-73"
      />
    </div>
  </ul>
</div>
`;

exports[`<CategorySubcategories /> renders without initial data 1`] = `
<div
  className="MuiPaper-root-1 MuiPaper-elevation2-5 MuiPaper-rounded-2"
>
  <div
    className="Component-root-31"
  >
    <h2
      className="MuiTypography-root-34 MuiTypography-title-40 Component-title-33"
    >
      Subcategories
    </h2>
    <div
      className="Component-action-28"
    >
      <button
        className="MuiButtonBase-root-64 MuiIconButton-root-58"
        disabled={false}
        onBlur={[Function]}
        onClick={[MockFunction]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex="0"
        type="button"
      >
        <span
          className="MuiIconButton-label-63"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root-67"
            color={undefined}
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
            />
          </svg>
        </span>
        <span
          className="MuiTouchRipple-root-73"
        />
      </button>
    </div>
  </div>
  <ul
    className="MuiList-root-80 MuiList-padding-81"
  >
    <li
      className="MuiListSubheader-root-101 MuiListSubheader-sticky-105"
    >
      No categories found
    </li>
  </ul>
</div>
`;
